tools/DSharpPlus.Tools.CodeBlockLanguageListGen/Program.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DSharpPlus.Tools.CodeBlockLanguageListGen;
+
+public static class Program
+{
+    private static readonly string TemplateFile;
+
+    static Program()
+    {
+        // Load the template file
+        // The filename could probably be hardcoded.
+        Assembly assembly = typeof(Program).Assembly;
+        foreach (string filename in assembly.GetManifestResourceNames())
+        {
+            if (!filename.EndsWith(".template", StringComparison.Ordinal) || assembly.GetManifestResourceStream(filename) is not Stream manifestStream)
+            {
+                continue;
+            }
+
+            TemplateFile = new StreamReader(manifestStream).ReadToEnd();
+            return;
+        }
+
+        throw new InvalidOperationException("Failed to load the template file.");
+    }
+
+    public static async Task Main()
+    {
+        // Grab the list of languages and their aliases
+        IReadOnlyList<string> languages = await GetLanguagesAsync();
+        StringBuilder languageListNode = new("new List<string>()\n");
```

Comment:
can we use a collection expression there?

Reply:
No. Collection expressions can only be used where `new()` can be used. Neither `new()` or `[]` know to create a list of strings